src/utils/seedRefHooks.ts

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+
+import { decryptDataRef, destroyDataRef } from 'utils/native';
+
+type TryCreateFunc = (encryptedSeed: string, password: string) => Promise<void>;
+type TryDestroyFunc = () => Promise<void>;
+
+type SeedRefHooks = [number, boolean, TryCreateFunc, TryDestroyFunc];
+
+export function useSeedRef(): SeedRefHooks {
+	const [dataRef, setDataRef] = useState<number>(0);
+	const [valid, setValid] = useState<boolean>(false);
+
+	// Decrypt a seed and store the reference. Must be called before signing.
+	const create: TryCreateFunc = function (encryptedSeed, password) {
+		return decryptDataRef(encryptedSeed, password).then((ref: number) => {
+			setDataRef(ref);
+			setValid(true);
+		});
+	};
+
+	// Destroy the decrypted seed. Must be called before this leaves scope or
+	// memory will leak.
+	const destroy: TryDestroyFunc = function () {
+		return destroyDataRef(dataRef).then(() => {
+			setValid(false);
+		});
+	};
+
+	return [dataRef, valid, create, destroy];
```

Comment:
```suggestion
    // Use the seed reference to sign a message. Will throw an error if
    // `destroy` has already been called or if `create` failed.
    const brainWalletSign: TrySign = function (message) {
        if (!valid) {
            // Seed reference was never created or was already destroyed.
            throw new Error('cannot sign with an invalid seed reference');
        }
        return EthkeyBridge.brainWalletSignWithRef(dataRef, message);
    }

    // Use a reference returned by decryptDataRef to sign a message
    const substrateSign: TrySign = (message) {
        if (!valid) {
            // Seed reference was never created or was already destroyed.
            throw new Error('cannot sign with an invalid seed reference');
        }
        return EthkeyBridge.substrateSignWithRef(dataRef, message);
    }
    
	return [dataRef, valid, create, destroy, brainWalletSign, substrateSign];
```

Reply:
Signing functions should be included to make sure `valid` is always checked before calling.